tests/spread/extensions/env-injector/task.yaml

```diff
@@ -0,0 +1,132 @@
+summary: Build and run a basic hello-world snap using extensions
+
+systems:
+  - ubuntu-24.04*
+
+environment:
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: "1"
+  SNAP_DIR: ../snaps/env-injector-hello
+  SNAP: env-injector-hello
+
+prepare: |
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.env
+  rm -f ./*.snap
+  rm -rf ./squashfs-root
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
+execute: |
+
+  check_env_var() {
+    local snap_app="$1"
+    local env_name="$2"
+    local exp_value="$3"
+
+    local actual_value
+
+    if ! eval "$snap_app" | grep -q "^${env_name}="; then
+      echo "Environment variable '$env_name' is not set."
+      return 1
+    fi
+
+    actual_value=$( "$snap_app" | grep "^${env_name}=" | cut -d'=' -f2-)
+
+    if [ "$actual_value" != "$exp_value" ]; then
+      echo "Environment variable '$env_name' does not match the expected value."
+      echo "Expected: '$env_name=$exp_value', but got: '$env_name=$actual_value'"
+      return 1
+    fi
+
+    return 0 
+  }
+
+  cd "$SNAP_DIR"
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=1 snapcraft
+
+  unsquashfs "${SNAP}"_1.0_*.snap
+
+
+  # Check that the env-exporter program is present
+  [ -f squashfs-root/bin/command-chain/env-exporter ]
+  
+  # Check that the exec-env script is present
+  [ -f squashfs-root/usr/bin/exec-env ]
+
+  snap install "${SNAP}"_1.0_*.snap --dangerous --classic
```

Comment:
> Why should this use classic confinement?

The confinement in which it is tested is irrelevant for these tests, since the extension is made to work on all confinements. Right?

Anyway, I've changed this to `devmode` installation.

Reply:
Here: https://github.com/canonical/snapcraft/pull/4925/commits/ab60705ffe6533e1d9d0afbdccbd75e8dfecd719 I tried to use the common directory, [but it fails anyway](https://github.com/canonical/snapcraft/actions/runs/10423760020/job/28871293025?pr=4925#step:4:352). 

So I reverted it to `classic` confinement, since it also satisfies the purpose of this test.

Reply:
The path is wrong. It should be /var/snap/<snap>/common/

https://snapcraft.io/docs/data-locations

Reply:
Also, the global file name on line 76 is different, missing the .env extension.

Classic confinement satisfies the purpose, but this is a special type of snap used here for the wrong purpose. By using a strictly confined snap the tests can verify that an env file can be read in confinement by the exporter.